src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/EnumValuesMustMatch.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    public class EnumValuesMustMatch : Rule
+    {
+        public override void Initialize(RuleRunnerContext context)
+        {
+            context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
+        }
+
+        private static bool isEnum(ITypeSymbol type)
+        {
+            return type != null && type.IsType && type.TypeKind == TypeKind.Enum;
+        }
+        private void RunOnTypeSymbol(ITypeSymbol left, ITypeSymbol right, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            if (!isEnum(left) || !isEnum(right))
+            {
+                return;
+            }
+            var leftMembers = left.GetMembers().Where(a => a.Kind == SymbolKind.Field).Select(a => ((IFieldSymbol)a)).ToDictionary(a => a.Name);
```

Comment:
NIT: we only use `var` when the type can be easily inferred from the right side of the assignment like `var regex = new Regex()`, but in all of this cases please use explicit types instead for readability